### PR TITLE
Added sectionBy func version to handle optionals

### PR DESCRIPTION
**Pull request by skytan4**

### CODE REVIEW

File: SwiftificationTests/ArrayTests.swift [hunk 1]
```diff
@@ -100,12 +100,14 @@ class ArrayTests: XCTestCase {
     }
     
     func testSectionBy() {
-        let array = [TestObject(name: "Test", type: 0), TestObject(name: "Test", type: 0), TestObject(name: "Test2", type: 0), TestObject(name: "Test2", type: 1), TestObject(name: "Test2", type: 1)]
+        let array = [TestObject(name: "Test", type: 0), TestObject(name: "Test", type: 0), TestObject(name: "Test2", type: 0), TestObject(name: "Test2", type: 1), TestObject(name: nil, type: 1),  TestObject(name: nil, type: 1)]
```
Comment (hiltonc) [inline SwiftificationTests/ArrayTests.swift]: Be sure to keep the test for the old method also.

**Comment by hiltonc** [comment]
👍

**Comment by hiltonc** [comment]
You'll probably want to up the version in the Podspec.

**Comment by nickmshelley** [comment]
I'm going to merge it because I don't want to wait for the tests, and I doubt the podspec change will break them... If it does then I'll fix it :)

**Comment by hiltonc** [comment]
Beat you.

**Comment by nickmshelley** [comment]
Oh nm, the merge button isn't there any more...

**Comment by hiltonc** [comment]
But you still have to fix it if it breaks.

**Pull request merged**